PublishedModules/glueckself/MpdClient/MpdClient.py

```diff
@@ -0,0 +1,109 @@
+from modules.MpdClient import mpdhelper
+
+from core.ProjectAliceExceptions import ModuleStartDelayed, ModuleStartingFailed
+from core.base.model.Intent import Intent
+from core.base.model.Module import Module
+from core.dialog.model.DialogSession import DialogSession
+from core.commons.commons import online
+
+#>>> mpd.status()
+#{'volume': '44', 'repeat': '0', 'random': '0', 'single': '0', 'consume': '0', 'playlist': '2', 'playlistlength': '13', 'mixrampdb': '0.000000', 'state': 'play', 'song': '3', 'songid': '4', 'time': '1:178', 'elapsed': '0.789', 'bitrate': '128', 'audio': '44100:24:2', 'nextsong': '4', 'nextsongid': '5'}
+
+# TODO have all intents enabled and answer if new state is not possible
+# e.g. "play some music" when already playing -> "music is already playing"
+# false matches?
+
+class MpdClient(Module):
+	"""
+	Author: glueckself
+	Description: Control an MPD server
+	"""
+
+	_INTENT_PLAY = Intent('mpdPlay')
+	_INTENT_STOP = Intent('mpdStop')
+	_INTENT_NEXT = Intent('mpdNext')
+	_INTENT_PREV = Intent('mpdPrev')
+
+	def __init__(self):
+		self._SUPPORTED_INTENTS = [
+			self._INTENT_PLAY,
+			self._INTENT_STOP,
+			self._INTENT_NEXT,
+			self._INTENT_PREV
+		]
+
+		super().__init__(self._SUPPORTED_INTENTS)
+		
+		self._host = self.getConfig('mpdHost')
+		self._port = self.getConfig('mpdPort')
+		self._password = self.getConfig('mpdPassword')
+		
+		self._mpdConnected = False
+		self._mpd = mpdhelper.MPDClient()
+		
+		if not self._host:
+			self._logger.warn(f'[{self.name}] MPD host not configured, not doing anything.')
+			return
+		
+		self.ThreadManager.doLater(interval=1, func=self._mpd_poll_status)
+	
+	def _mpd_poll_status(self):
+		status = self._mpd.status()
+		if not status:
+			self._mpdConnected = False
+			self._mpd.disconnect()
+			self._connect()
+			self.ThreadManager.doLater(interval=1, func=self._mpd_poll_status)
+			return
+
+		self._mpdConnected = True
+		playbackStatus = False
+		if status['state'] == 'play':
+			playbackStatus = True
+		
+		try:
+			intents = list()
+			# when playing: stop, next, prev. when stopped: play
+			intents = [
+				{'intentId': 'mpdPlay', 'enable': not playbackStatus },
+				{'intentId': 'mpdStop', 'enable': playbackStatus },
+				{'intentId': 'mpdNext', 'enable': playbackStatus },
+				{'intentId': 'mpdPrev', 'enable': playbackStatus }
+			]
+			self.MqttManager.configureIntents(intents)
```

Comment:
Well done, this is the way to go

Reply:
This doesn't work, Snips NLU has some errors about the intent id but I'm not sure where they are coming from (the intents are listed in the Snips console). Maybe the id is something else, ModuleManager uses `'intentId': intent.justTopic if hasattr(intent, 'justTopic') else intent`

Reply:
Workaround, I use always-on-intents now.
While the workaround is better for my module, this may be useful for other modules and should be looked into I think (maybe a low priority issue?)